pkg/commands/cmdlib/bundleio.go

```diff
@@ -29,88 +29,218 @@ import (
 	}
 
 	// For now, we can only inline component data files because we need the path
 	// context.
-	if (g.InlineComponents || g.InlineObjects) && g.BundleFile != "" {
-		return inlineData(ctx, b, rw, g)
+	if (g.InlineComponents || g.InlineObjects) && g.InputFile != "" {
+		return brw.inlineData(ctx, bw, g)
 	}
-	return b, nil
-}
-
-func convertBundle(ctx context.Context, bt []byte, g *GlobalOptions) (*bundle.Bundle, error) {
-	return converter.FromContentType(g.InputFormat, bt).ToBundle()
+	return bw, nil
 }
 
 // inlineData inlines a cluster bundle before processing
-func inlineData(ctx context.Context, data *bundle.Bundle, rw files.FileReaderWriter, g *GlobalOptions) (*bundle.Bundle, error) {
-	inliner := inline.NewInlinerWithScheme(
-		files.FileScheme,
-		&files.LocalFileObjReader{filepath.Dir(g.BundleFile), rw},
-		inline.DefaultPathRewriter,
-	)
+func (brw *BundleReaderWriter) inlineData(ctx context.Context, bw *BundleWrapper, g *GlobalOptions) (*BundleWrapper, error) {
 	var err error
-	if g.InlineComponents {
-		data, err = inliner.InlineBundleFiles(ctx, data)
+	inliner := brw.makeInlinerFn(brw.rw, g.InputFile)
+	if g.InlineComponents && bw.Bundle != nil {
+		bw.Bundle, err = inliner.InlineBundleFiles(ctx, bw.Bundle)
 		if err != nil {
 			return nil, fmt.Errorf("error inlining component data files: %v", err)
 		}
 	}
-	if g.InlineObjects {
-		data, err = inliner.InlineComponentsInBundle(ctx, data)
+	if g.InlineObjects && bw.Bundle != nil {
+		bw.Bundle, err = inliner.InlineComponentsInBundle(ctx, bw.Bundle)
 		if err != nil {
 			return nil, fmt.Errorf("error inlining objects: %v", err)
 		}
 	}
-	return data, nil
+	if g.InlineObjects && bw.Component != nil {
+		bw.Component, err = inliner.InlineComponent(ctx, bw.Component)
+		if err != nil {
+			return nil, fmt.Errorf("error inlining objects: %v", err)
+		}
+	}
+	return bw, nil
 }
 
-// WriteStructuredContents writes some structured contents from some object `obj`. The contents must
-// be serializable to both JSON and YAML.
-func WriteStructuredContents(ctx context.Context, obj interface{}, rw files.FileReaderWriter, g *GlobalOptions) error {
-	bytes, err := converter.FromObject(obj).ToContentType(g.OutputFormat)
+// WriteBundleData writes either the component or bundle object from the BundleWrapper.
+func (brw *BundleReaderWriter) WriteBundleData(ctx context.Context, bw *BundleWrapper, g *GlobalOptions) error {
+	if bw.Bundle != nil {
+		return brw.WriteStructuredContents(ctx, bw.Bundle, g)
+	} else if bw.Component != nil {
+		return brw.WriteStructuredContents(ctx, bw.Component, g)
+	} else {
+		return fmt.Errorf("both bundle and component were nil")
+	}
+}
+
+// WriteStructuredContents writes some structured contents from some object
+// `obj`. The contents must be serializable to both JSON and YAML.
+func (brw *BundleReaderWriter) WriteStructuredContents(ctx context.Context, obj interface{}, g *GlobalOptions) error {
+	outFmt := g.OutputFormat
```

Comment:
Is the file extension overriding the flag?

Reply:
Yeah, it is. Updated based on your comment below.